cumulusci/cli/cci.py

```diff
@@ -1425,83 +1425,235 @@ def task_info(runtime, task_name):
+        name, None otherwise."""
+        opt_names = sorted(self._parse_option_names(args))
+        for i, name in enumerate(opt_names):
+            if i == len(opt_names) - 1:
+                break
+            elif name == opt_names[i + 1]:
+                return name
+
+        return None
+
+    def _parse_option_names(self, args):
+        """
+        Given a list of arguments, parse
+        out the provided option names.
+        new syntax: --name value
+        old syntax: -o name value
+        """
+        opt_names = []
+        for i, opt in enumerate(args):
+            if opt == "-o":
+                opt_names.append(args[i + 1])
+            elif opt.startswith("--"):
+                opt_names.append(opt.strip("-"))
+
+        return opt_names
+
+    def _option_in_task(self, opt_name, task_name):
+        """
+        Returns True if opt_name is the name of an
+        option defined in the given task, else False.
+        """
+        task = RUNTIME.project_config.get_task(task_name)
+        task_class = import_global(task.config["class_path"])
+
+        if "task_options" not in task_class.__dict__.keys():
+            task_option_names = self._get_task_options_in_hierarchy(task_class)
+        else:
+            task_option_names = task_class.__dict__["task_options"].keys()
+
+        return opt_name in task_option_names
+
+    def _get_task_options_in_hierarchy(self, task_class):
+        """
+        Given a task class, search all base classes
+        for the first defines `task_options`.
+
+        Returns:
+            A list of the option names
+
+        Raises:
+            CumulusCIException if `task_options` are not defined
+        """
+        for base in task_class.__bases__:
+            if hasattr(base, "task_options"):
+                return [name for name in base.task_options.keys()]
+
+        raise CumulusCIException(
+            f"Unable to find `task_options` in base classes:\n{task_class.__bases__}"
+        )
+
+    def _get_click_options_for_task(self, task_options):
+        """
+        Given a dict of options in a task, constructs and returns the
+        corresponding list of click.Option instances
+        """
+        click_options = []
+        if task_options:
+            for name, properties in task_options.items():
+                # NOTE: When task options aren't explicitly given via the command line
+                # click complains that there are no values for options. We set required=False
+                # to mitigate this error. Task option validation should be performed at the
+                # task level via task._validate_options() or Pydantic models.
+                click_options.append(
+                    click.Option(
+                        param_decls=(f"--{name}",),
+                        required=False,  # don't enforce option values in Click
+                        help=properties.pop("description", ""),
```

Comment:
Don't use pop! This is probably mutating the task config within the actual project_config!